radar/radar.go

```diff
@@ -120,28 +126,21 @@ func (radar *Radar) scan(logger lager.Logger, resourceName string) error {
 		return nil
 	}
 
-	config, _, err := radar.db.GetConfig()
+	savedResource, err := radar.db.GetResource(resourceName)
 	if err != nil {
-		logger.Error("failed-to-get-config", err)
-		// don't propagate error; we can just retry next tick
-		return nil
+		return err
 	}
 
-	resourceConfig, found := config.Resources.Lookup(resourceName)
-	if !found {
-		logger.Info("resource-removed-from-configuration")
-		// return an error so that we exit
-		return resourceNotConfiguredError{ResourceName: resourceName}
+	if savedResource.Paused {
+		return nil
 	}
 
-	savedResource, err := radar.db.GetResource(resourceName)
+	resourceConfig, err := radar.getResourceConfig(logger, resourceName)
 	if err != nil {
 		return err
 	}
 
-	if savedResource.Paused {
-		return nil
-	}
+	defer radar.configureTimer(resourceConfig.CheckEvery)
```

Comment:
@vito I defer so that I don't reconfigure the timer until the whole function has finished.  Are you sure you want the behavior where the timer gets created and starts before that initial check has finished?